airgun/entities/contenthost.py

```diff
@@ -55,27 +59,70 @@ def execute_package_action(self, entity_name, action_type, value):
         view.progressbar.wait_for_result()
         return view.read()
 
+    def execute_module_stream_action(self, entity_name, action_type, module_name, stream_version,
+                                     customize=False, customize_values=None):
+        """Execute remote module_stream action on a content
+
+        :param entity_name: content host name to remotely execute package
+            action on
+        :param action_type: remote action to execute on content host. Action value can be one of
+            them e.g. 'Enable', 'Disable', 'Install', 'Update', 'Remove', 'Reset'
+        :param module_name: Module Stream name to remotely
+            install/upgrade/remove (depending on `action_type`)
+        :param stream_version: this uniquely identifies the module with Stream Version
+        :param customize: special action type which should call on content host module streams
+            tab to run some custom expression related to module streams
+        :param customize_values: need to pass if run some custom command expression in content host
+
+        :return: Returns a dict containing job status details
+        """
+        if customize_values is None:
+            customize_values = {}
+        view = self.navigate_to(self, 'Edit', entity_name=entity_name)
+        view.module_streams.search('name = {} and stream = {}'.format(module_name, stream_version))
+        action_type = dict(is_customize=customize, action=action_type)
+        view.module_streams.table.row(
+            name=module_name, stream=stream_version)['Actions'].fill(action_type)
+        if customize:
+            view = JobInvocationCreateView(view.browser)
+            view.fill(customize_values)
+            view.submit.click()
+        view = JobInvocationStatusView(view.browser)
+        view.wait_for_result()
+        return view.read()
+
     def search_package(self, entity_name, package_name):
         """Search for specific package installed in content host"""
         view = self.navigate_to(self, 'Edit', entity_name=entity_name)
         view.packages_installed.search(package_name)
         return view.packages_installed.table.read()
 
-    def install_errata(self, entity_name, errata_id):
+    def search_module_stream(self, entity_name, module_name, status='All'):
+        """Search for specific package installed in content host"""
+        view = self.navigate_to(self, 'Edit', entity_name=entity_name)
+        view.module_streams.search(module_name, status)
+        return view.module_streams.table.read()
+
+    def install_errata(self, entity_name, errata_id, install_via=None):
         """Install errata on a content host
 
         :param name: content host name to apply errata on
         :param errata_id: errata id or title, e.g. 'RHEA-2012:0055'
-
+        :param str install_via: Via which mean to install errata. Available
+        options: "via Katello Agent", "via remote execution",
+        "via remote execution - customize first"
```

Comment:
I thought that there could a lot of tests, where have to pass some string to let know that they wanted to run using via katello agent, to avoid that I have used the None. 

As per your view, I will have to update the logic and even all tests which consuming this?

Reply:
Defaulting to Katello Agent is fine. We definitely don't want to update all existing tests only because new parameter is added.

What I mean is the value that caller must pass - right now it is the same as string in UI. If string in UI changes, we will have to update tests. I would rather have caller use one of stable, well-defined names that is translated to UI string in this method. Then if string in UI changes, we will have to update only this one place.

If we already pass UI strings in Robottelo, could you link some example of that? I may think it's bad idea, but if it's already done, then probably this is not blocking.

Reply:
@mirzal now I understand your point, I can do that, will be good practice too. I will update and compare the string as we will not have to depend on the test to pass locator instead taken care at business/airgun level

Reply:
@mirzal I just come to know why I did like that e.g. 
https://github.com/SatelliteQE/airgun/blob/fb30886d1a6f6290c6cb8886ee9c5e87e226dceb/airgun/entities/hostcollection.py#L120
https://github.com/SatelliteQE/airgun/blob/fb30886d1a6f6290c6cb8886ee9c5e87e226dceb/airgun/entities/hostcollection.py#L148